hyperspy/_signals/lazy.py

```diff
@@ -1013,6 +1014,100 @@ def post(a):
+        computationally efficient approach to compute the navigator. The data
+        can be rechunk by specifying the ``chunks_number`` argument.
+
+        Parameters
+        ----------
+        index : (int, float, None) or iterable, optional
+            Specified where to take the sum, follows HyperSpy indexing syntax
+            for integer and float. If None, the index is the centre of the
+            signal_space
+        chunks_number : (int, None) or iterable, optional
+            Define the number of chunks in the signal space used for rechunk
+            the when calculating of the navigator. Useful to define the range
+            over which the sum is calculated.
+            If None, the existing chunking will be considered when picking the
+            chunk used in the navigator calculation.
+        %s
+
+        Returns
+        -------
+        None.
+
+        Note
+        ----
+        The number of chunks will affect where the sum is taken. If the sum
+        needs to be taken in the centre of the signal space (for example, in
+        the case of diffraction pattern), the number of chunk needs to be an
+        odd number, so that the middle is centered.
+
+        """
+
+        signal_shape = self.axes_manager.signal_shape
+
+        if index is None:
+            index = [round(shape / 2) for shape in signal_shape]
+        else:
+            if not isiterable(index):
+                index = [index] * len(signal_shape)
+            index = [axis._get_index(_idx)
+                     for _idx, axis in zip(index, self.axes_manager.signal_axes)]
+        _logger.info(f"Using index: {index}")
+
+        if chunks_number is None:
+            chunks = self.data.chunks
+        else:
+            if not isiterable(chunks_number):
+                chunks_number = [chunks_number] * len(signal_shape)
+            # Determine the chunk size
+            signal_chunks = da.core.normalize_chunks(
+                [int(size / cn) for cn, size in zip(chunks_number, signal_shape)],
+                shape=signal_shape
+                )
+            # Needs to reverse the chunks list to match dask chunking order
+            signal_chunks = list(signal_chunks)[::-1]
+            navigation_chunks = ['auto'] * len(self.axes_manager.navigation_shape)
+            chunks = self.data.rechunk([*navigation_chunks, *signal_chunks],
+                                       balance=True).chunks
+
+        # Get the slice of the corresponding chunk
+        signal_size = len(signal_shape)
+        signal_chunks = tuple(chunks[i-signal_size] for i in range(signal_size))
+        _logger.info(f"Signal chunks: {signal_chunks}")
+        isig_slice = get_signal_chunk_slice(index, chunks)
+
+        _logger.info(f'Computing sum over signal dimension: {isig_slice}')
+        axes = [axis.index_in_array for axis in self.axes_manager.signal_axes]
+        navigator = self.isig[isig_slice].sum(axes)
```

Comment:
I will be inclined to stick to the default here: through `rechunk=True` in `sum`, it will defer the rechunking to `dask` and if this doesn't play well, then it means that either we are doing something or dask is doing something, in which case we should investigate and figure out. Unless, it can be made clear that this is a bad, I would suggest to stick to the defaults. What do you think?

Reply:
I am not really sure to be honest. My experience after working with dask quite a bit, is that rechunking datasets which already has a reasonable chunk size just slows things down. Keeping the chunks "intact" from the file all the way through the processing is typically the best. However, this depends on what happens with the dask array after the rechunking: if it is computed right away (like in this case), it is typically fine.

So in this case, using the default (`rechunk=True`) should be ok.